### PR TITLE
Python: MWE for call-graph tracing and ql comparison

### PR DESCRIPTION
**Pull request by RasmusWL**
I'm not sure whether we want to merge this in now or not. I'll let that be up to my reviewer. If not, I'll just convert this into a draft PR until it is more complete.

### CODE REVIEW

File: python/tools/recorded-call-graph-metrics/README.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Recorded Call Graph Metrics
+
+also known as _call graph tracing_.
+
+Execute a python program and for each call being made, record the call and callable. This allows us to compare call graph resolution from static analysis with actual data -- that is, can we statically determine the target of each actual call correctly.
+
+This is still in the early stages, and currently only support a very minimal working example (to show that this approach might work).
```
Comment (intrigus-lgtm) [inline python/tools/recorded-call-graph-metrics/README.md:7]:
```suggestion
This is still in the early stages, and currently only supports a very minimal working example (to show that this approach might work).
```
I might be wrong here given that English is not my first language.
Or should it be "and we currently only ..."?

Comment (RasmusWL) [reply python/tools/recorded-call-graph-metrics/README.md]: no that sounds about right to me as well :+1:

File: python/tools/recorded-call-graph-metrics/cg_trace.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+
+
+class CSVExporter(Exporter):
+
+    @staticmethod
+    def export(recorded_calls, outfile_path):
+        with open(outfile_path, 'w', newline='') as csv_file:
+            writer = None
+            for (call, callable) in recorded_calls:
+
```
Comment (tausbn) [inline python/tools/recorded-call-graph-metrics/cg_trace.py:131]: This empty line is a bit strange. Was it introduced by an autoformatter? :thinking:

Comment (RasmusWL) [reply python/tools/recorded-call-graph-metrics/cg_trace.py:131]: nope. that was just me :)

File: python/tools/recorded-call-graph-metrics/cg_trace.py [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+            inst_index = frame.f_lasti,
+        )
+
+
+@dataclasses.dataclass(frozen=True)
+class Callable():
+    """A callable (Function/Lambda) should (hopefully) be uniquely identified by its name and
+    location (filename+line number)
+
+    TODO: Callable is maybe not a good name, since classes with __call__ will return true
```
[the remark this answers is not in the document]
Comment (RasmusWL) [reply python/tools/recorded-call-graph-metrics/cg_trace.py:66]: let's discuss when we talk later today :)

**Pull request merged**